app/js/app.js

```diff
@@ -0,0 +1,172 @@
+    var rect = c.getBoundingClientRect();
+    var x = Math.round((e.clientX-rect.left)/(rect.right-rect.left)*c.width);
+    var y = Math.round((e.clientY-rect.top)/(rect.bottom-rect.top)*c.height);
+
+    var touch = e.touches[0];
+    var mouseEvent = new MouseEvent("mousedown", {
+        clientX: touch.clientX,
+        clientY: touch.clientY
+    });
+    c.dispatchEvent(mouseEvent);
+}, false);
+
+c.addEventListener("touchmove", function(e){
+    var touch = e.touches[0];
+    var mouseEvent = new MouseEvent("mousemove", {
+        clientX: touch.clientX,
+        clientY: touch.clientY
+    });
+    c.dispatchEvent(mouseEvent);
+}, false);
+
+c.addEventListener("touchend", function (e){
+    var mouseEvent = new MouseEvent("mouseup", {});
+    c.dispatchEvent(mouseEvent);
+}, false);
+
+//desktop code
+c.onmousedown = function(e){
+	e.preventDefault();
+
+    var rect = c.getBoundingClientRect();
+    var x = Math.round((e.clientX-rect.left)/(rect.right-rect.left)*c.width);
+    var y = Math.round((e.clientY-rect.top)/(rect.bottom-rect.top)*c.height);
+
+	isDrawing = true;
+	ctx.lineWidth = 3;
+  	ctx.lineJoin = ctx.lineCap = 'round';
+  	ctx.moveTo(x, y);
+}
+
+c.onmousemove = function(e) {
+	e.preventDefault();
+    
+	 var rect = c.getBoundingClientRect();
+	 var x = Math.round((e.clientX-rect.left)/(rect.right-rect.left)*c.width);
+	 var y = Math.round((e.clientY-rect.top)/(rect.bottom-rect.top)*c.height);
+	if(isDrawing){
+		ctx.lineTo(x, y);
+    	ctx.stroke();
+	}
+}
+
+c.onmouseup = function(e) {
+	e.preventDefault();
+	isDrawing = false;
+}
+
+
+//timer logic
+
+
+
+function getTimeRemaining(endtime) {
+  var t = Date.parse(endtime) - Date.parse(new Date());
+  var seconds = Math.floor((t / 1000) % 60);
+  var minutes = Math.floor((t / 1000 / 60) % 60);
+  var hours = Math.floor((t / (1000 * 60 * 60)) % 24);
+  var days = Math.floor(t / (1000 * 60 * 60 * 24));
+  return {
+    'total': t,
+    'minutes': minutes,
+    'seconds': seconds
+  };
+}
+
+function initializeClock(id, endtime) {
+  var clock = document.getElementById('timer');
+  // var daysSpan = clock.querySelector('.days');
+  // var hoursSpan = clock.querySelector('.hours');
+  var minutesSpan = clock.querySelector('.minutes');
+  var secondsSpan = clock.querySelector('.seconds');
+
+  function updateClock() {
+    var t = getTimeRemaining(endtime);
+
+    var minutesSpan = document.getElementById('minutes');
+    var secondsSpan = document.getElementById('seconds');
+    minutesSpan.innerHTML = ('0' + t.minutes).slice(-2);
+    secondsSpan.innerHTML = ('0' + t.seconds).slice(-2);
+
+    if (t.total <= 0) {
+      clearInterval(timeinterval);
+    }
+  }
+
+  updateClock();
+  var timeinterval = setInterval(updateClock, 1000);
+  console.log("clocking in");
+}
+
+//var deadline = new Date(Date.parse(new Date()) + 15 * 24 * 60 * 60 * 1000);
+var deadline = new Date(Date.parse(new Date()) + 2 * 60 * 1000);
+initializeClock('timer', deadline);
+
+/**var d = Date();
```

Comment:
Prefer to keep commented code out of the master branch. You could put this in another branch if you want to keep working on it